### PR TITLE
feat: support masking and get portal login access token

### PR DESCRIPTION
**Pull request by TristanSpeakEasy**

### CODE REVIEW

**Review by alexadrake — APPROVED** [review]
Looks good to me

File: src/main/java/dev/speakeasyapi/sdk/masking/Masking.java [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    private Map<String, String> requestHeaderMasks = new HashMap<>();
+    private Map<String, String> responseHeaderMasks = new HashMap<>();
+    private Map<String, String> requestCookieMasks = new HashMap<>();
+    private Map<String, String> responseCookieMasks = new HashMap<>();
+    private Map<String, String> requestBodyMasksString = new HashMap<>();
+    private Map<String, String> responseBodyMasksString = new HashMap<>();
+    private Map<String, String> requestBodyMasksNumber = new HashMap<>();
+    private Map<String, String> responseBodyMasksNumber = new HashMap<>();
+
+    public static class Builder {
```
[the remark this answers is not in the document]
Comment (TristanSpeakEasy) [reply src/main/java/dev/speakeasyapi/sdk/masking/Masking.java:28]: was actually based on a blog post I was reading about the builder pattern and they had it as static, I blindly followed tbh I will look if there is any impact to changing this

File: src/main/java/dev/speakeasyapi/sdk/models/SpeakeasyAccessTokenFilterBuilder.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package dev.speakeasyapi.sdk.models;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import dev.speakeasyapi.accesstokens.Embedaccesstoken.EmbedAccessTokenRequest.Filter;
+
+public class SpeakeasyAccessTokenFilterBuilder {
```
[the remark this answers is not in the document]
Comment (TristanSpeakEasy) [reply src/main/java/dev/speakeasyapi/sdk/models/SpeakeasyAccessTokenFilterBuilder.java:10]: For the time being, I left it as it does provide now some good abstraction away from the request, as I am using the same request for two different controller methods and the original naming might have been confusing

File: src/main/java/dev/speakeasyapi/sdk/masking/Masking.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (TristanSpeakEasy) [reply src/main/java/dev/speakeasyapi/sdk/masking/Masking.java:28]: ah figured it out, as I instantiate it with a static method that is why

File: docs/micronaut.md [hunk 3]
```diff
@@ -184,23 +184,23 @@ Below are some examples on how to generate access tokens:
 
 ```java
 @Get("/embed_access_token")
-public String getSpeakeasyEmbedAccessToken(@RequestAttribute(SpeakeasyMiddlewareController.Key) SpeakeasyMiddlewareController ctrl){
+ublic String getSpeakeasyEmbedAccessToken(@RequestAttribute(SpeakeasyMiddlewareController.Key) SpeakeasyMiddlewareController controller){
```
[the remark this answers is not in the document]
Comment (TristanSpeakEasy) [reply docs/micronaut.md]: fixed

**Pull request merged**